extensions/ql-vscode/src/distribution.ts

```diff
@@ -553,21 +554,28 @@ export class ReleasesApiConsumer {
   private static readonly _maxRedirects = 20;
 }
 
-export async function extractZipArchive(archivePath: string, outPath: string): Promise<void> {
-  const archive = await unzipper.Open.file(archivePath);
-  await archive.extract({
-    concurrency: 4,
-    path: outPath
-  });
-  // Set file permissions for extracted files
-  await Promise.all(archive.files.map(async file => {
-    // Only change file permissions if within outPath (path.join normalises the path)
-    const extractedPath = path.join(outPath, file.path);
-    if (extractedPath.indexOf(outPath) !== 0 || !(await fs.pathExists(extractedPath))) {
-      return Promise.resolve();
-    }
-    return fs.chmod(extractedPath, file.externalFileAttributes >>> 16);
-  }));
+export async function extractArchive(archivePath: string, outPath: string): Promise<void> {
+  if (archivePath.endsWith('zip')) {
+    const archive = await unzipper.Open.file(archivePath);
+    await archive.extract({
+      concurrency: 4,
+      path: outPath
+    });
+    // Set file permissions for extracted files
+    await Promise.all(archive.files.map(async file => {
+      // Only change file permissions if within outPath (path.join normalises the path)
+      const extractedPath = path.join(outPath, file.path);
+      if (extractedPath.indexOf(outPath) !== 0 || !(await fs.pathExists(extractedPath))) {
+        return Promise.resolve();
+      }
+      return fs.chmod(extractedPath, file.externalFileAttributes >>> 16);
+    }));
+  } else {
+    await tar.extract({
+      file: archivePath,
+      cwd: outPath,
+    });
```

Comment:
Hmm, I'm slightly surprised that we only need to set file permissions when we extract zip archives.

Reply:
My understanding is that `.zip` files don't typically store file permissions so the information is lost after zipping and unzipping (although with the right tooling it is _possible_ to preserve the permissions), whereas `.tar.gz` ones typically do.